test-violations.py

```diff
@@ -0,0 +1,62 @@
+from attrs import define
+from attrs import field
+
+
+@attr.s(auto_attribs=True)
+class ExampleClass(object):
+    # ruleid:attr-mutable-initializer
+    empty_dict = {}
+    # ruleid:attr-mutable-initializer
+    empty_list = []
+    # ruleid:attr-mutable-initializer
+    somedict = dict()
+    # ruleid:attr-mutable-initializer
+    somelist = list()
+    # ruleid:attr-mutable-initializer
+    someset = set()
+    # ok:attr-mutable-initializer
+    ex_good1 = attr.ib(factory=dict)
+    # ok:attr-mutable-initializer
+    ex_good2 = attr.ib(factory=list)
+    # ok:attr-mutable-initializer
+    ex_good3 = attr.ib(factory=set)
+    # ruleid:attr-mutable-initializer
+    myset = {1, 2, 3}
```

Comment:
:exclamation:**Cycode**: SAST violation: 'Unsafe usage of mutable initializer with attr.s decorator. Multiple instances of this class will re-use the same data structure, which is likely not the desired behavior. Consider instead: replace assignment to mutable initializer (ex. dict() or {}) with attr.ib(factory=type) where type is dict, set, or list'.
 
Severity: Medium
 <!-- b00b4090-287d-4e9f-8e87-9b820271a116 -->